modules/gapi/test/common/gapi_imgproc_tests_common.hpp

```diff
@@ -0,0 +1,52 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#ifndef OPENCV_GAPI_IMGPROC_TESTS_COMMON_HPP
+#define OPENCV_GAPI_IMGPROC_TESTS_COMMON_HPP
+
+#include "gapi_tests_common.hpp"
+#include "../../include/opencv2/gapi/imgproc.hpp"
+
+#include <opencv2/imgproc.hpp>
+
+namespace opencv_test
+{
+template<typename G_In, typename In>
```

Comment:
You can use g_type_of_t from gtype_traits.hpp to automatically deduce G_In type

Reply:
Ruslan, thank you so much! This is exactly what I was looking for here! So much better with this